onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -215,33 +254,51 @@ Status LayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
     // Get the inputs for the new LayerNormalization node.
     NodeArg* scale = nullptr;
     NodeArg* bias = nullptr;
+
+    // scale and bias could be multi-dims; we only support it for training at the moment
+    // because SkipLayerNorm kernel, for example, has dependency on single dim size
     for (size_t i = 0; i < mul_node.MutableInputDefs().size(); i++) {
       if (graph_utils::NodeArgIsConstant(graph, *(mul_node.MutableInputDefs()[i])) ||
           graph_utils::IsGraphInput(graph, mul_node.MutableInputDefs()[i])) {
+#ifdef ENABLE_TRAINING
+        if (axes_values.empty() || mul_node.MutableInputDefs()[i]->Shape()->dim_size() == axes_values.size()) {
+          scale = mul_node.MutableInputDefs()[i];
+        }
+#elif
         // Scale must be 1d.
         if (mul_node.MutableInputDefs()[i]->Shape()->dim_size() == 1) {
           scale = mul_node.MutableInputDefs()[i];
         }
+#endif
       }
     }
 
     for (size_t i = 0; i < last_add_node.MutableInputDefs().size(); i++) {
       if (graph_utils::NodeArgIsConstant(graph, *(last_add_node.MutableInputDefs()[i])) ||
           graph_utils::IsGraphInput(graph, last_add_node.MutableInputDefs()[i])) {
+#ifdef ENABLE_TRAINING
+        if (axes_values.empty() || last_add_node.MutableInputDefs()[i]->Shape()->dim_size() == axes_values.size()) {
+          bias = last_add_node.MutableInputDefs()[i];
+        }
+#elif
         // Bias must be 1d.
         if (last_add_node.MutableInputDefs()[i]->Shape()->dim_size() == 1) {
           bias = last_add_node.MutableInputDefs()[i];
         }
+#endif
       }
     }
     if (scale == nullptr || bias == nullptr) {
       continue;
     }
 
     // Scale and bias must have the same dimension.
-    if (scale->Shape()->dim(0).dim_value() != bias->Shape()->dim(0).dim_value()) {
-      continue;
+    for (int i = 0; i < scale->Shape()->dim_size(); i++) {
+      if (scale->Shape()->dim(i).dim_value() != bias->Shape()->dim(i).dim_value()) {
+        continue;
```

Comment:
the `continue` here is not able to proceed onto the outer loop, with the new loop that is wrapped around it.